### PR TITLE
[Develop] CI/CD support for Catalina

### PR DESCRIPTION
**Pull request by NorseGaud**
## Change Description

Added new Catalina 10.15.1 to the Buildkite CI pipelines.

We also tested that a Mojave created blockchain will migrate without issue to a catalina built nodeos instance.

**Mojave is skipped in all but a post-merge situation for protected branches**:
- PINNED Forked branch/PR example: https://buildkite.com/EOSIO/eosio/builds/20233 : https://github.com/EOSIO/eos/pull/8532
- LRT: https://buildkite.com/EOSIO/eosio-lrt/builds/992
- UNPINNED: https://buildkite.com/EOSIO/eosio-build-unpinned/builds/1234

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: .cicd/generate-pipeline.sh [hunk 1]
```diff
@@ -643,6 +655,29 @@ cat <<EOF
     timeout: ${TIMEOUT:-10}
     skip: ${SKIP_MACOS_10_14}${SKIP_PACKAGE_BUILDER}${SKIP_MAC}
 
+  - label: ":darwin: macOS 10.15 - Package Builder"
+    command:
+      - "git clone \$BUILDKITE_REPO eos && cd eos && $GIT_FETCH git checkout -f \$BUILDKITE_COMMIT"
+      - "cd eos && buildkite-agent artifact download build.tar.gz . --step ':darwin: macOS 10.15 - Build' && tar -xzf build.tar.gz"
+      - "cd eos && ./.cicd/package.sh"
+    plugins:
+      - chef/anka#v0.5.4:
```
Comment (jeffreyssmith2nd) [inline .cicd/generate-pipeline.sh]: `EOSIO/anka#v0.5.7` ?

File: .cicd/platforms/pinned/macos-10.15-pinned.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
```
Comment (jeffreyssmith2nd) [inline .cicd/platforms/pinned/macos-10.15-pinned.sh]: I assume this file is very similar to the Mojave file?

Comment (NorseGaud) [reply .cicd/platforms/pinned/macos-10.15-pinned.sh]: Yep

File: .cicd/generate-pipeline.sh (hunk 1 above)
Comment (NorseGaud) [reply .cicd/generate-pipeline.sh]: Good catch!

**Pull request merged**